src/Auth/CookieAuthenticate.php

```diff
@@ -0,0 +1,169 @@
+ *            ]
+ *        ]
+ *    ]);
+ * ```
+ */
+class CookieAuthenticate extends BaseAuthenticate
+{
+
+    /**
+     * Constructor
+     *
+     * @param \Cake\Controller\ComponentRegistry $registry The Component registry
+     *   used on this request.
+     * @param array $config Array of config to use.
+     */
+    public function __construct(ComponentRegistry $registry, $config)
+    {
+        $this->_registry = $registry;
+
+        $this->config([
+            'cookie' => [
+                'name' => 'RememberMe',
+                'expires' => '+2 weeks',
+            ],
+            'crypt' => 'aes',
+            'fields' => ['tokenCreated' => null]
+        ]);
+
+        $aliases = [
+            'identifier' => 'username',
+            'token' => 'password'
+        ];
+
+        foreach ($aliases as $alias => $field) {
+            if (isset($config['fields'][$alias])) {
+                $config['fields'][$field] = $config['fields'][$alias];
+                unset($config['fields'][$alias]);
+            }
+        }
+        $this->config($config);
+    }
+
+    /**
+     * Authenticates the identity contained in the cookie.  Will use the
+     * `userModel` config, and `fields` config to find COOKIE data that is used
+     * to find a matching record in the model specified by `userModel`. Will return
+     * false if there is no cookie data, either identifier or token is missing,
+     * or if the scope conditions have not been met.
+     *
+     * @param Request $request The unused request object.
+     * @return mixed False on login failure. An array of User data on success.
+     * @throws \RuntimeException If CookieComponent is not loaded.
+     */
+    public function getUser(Request $request)
+    {
+        if (!isset($this->_registry->Cookie) ||
+        !$this->_registry->Cookie instanceof CookieComponent
+        ) {
+            throw new MissingComponentException(['class' => 'CookieComponent']);
+        }
+
+        $cookieConfig = $this->_config['cookie'];
+        $cookieName = $this->_config['cookie']['name'];
+        unset($cookieConfig['name']);
+        $this->_registry->Cookie->configKey($cookieName, $cookieConfig);
+
+        $data = $this->_registry->Cookie->read($cookieName);
+        if (empty($data)) {
+            return false;
+        }
+
+        extract($this->_config['fields']);
+
+        if (empty($data[$username]) || empty($data[$password])) {
+            return false;
+        }
+
+        $user = $this->_findUser($data[$username], $data[$password]);
+
+        if ($tokenCreated !== null) {
+            $expiration = str_replace('+', '', $this->config('cookie.expires'));
+            if (!$user[$tokenCreated]->wasWithinLast($expiration)) {
+                return false;
+            }
+
+            if (isset($user['password'])) {
+                unset($user['password']);
+            }
+        }
+
+        if ($user) {
+            $request->session()->write(
```

Comment:
Don't write to session. Just return the record and AuthComponent will write it to configured storage class.